src/engine/shared/console.cpp

```diff
@@ -155,7 +155,16 @@ int CConsole::ParseArgs(CResult *pResult, const char *pFormat)
 				pResult->AddArgument(pStr);
 
 				if(Command == 'r') // rest of the string
+				{
+					// drop trailing whitespace
+					char *pEnd = pStr + str_length(pStr) - 1;
```

Comment:
This should probably be done in StrVariableCommand so it just effects the string config values and other executable commands keep the old behaviour. And it could check for utf8 whitespace in general.

Reply:
I don't think that it should just affect `StrVariableCommand`. E.g. it would be weird if `change_map` had different semantics from `sv_map`.

Reply:
Some commands could rely on it, maybe it's best to keep the old behaviour.

Reply:
The following is a list of all commands that use the `r` argument.
```
src/game/server/gamecontext.cpp:	Console()->Register("change_map", "?r", CFGFLAG_SERVER|CFGFLAG_STORE, ConChangeMap, this, "Change map");
src/game/server/gamecontext.cpp:	Console()->Register("say", "r", CFGFLAG_SERVER, ConSay, this, "Say in chat");
src/game/server/gamecontext.cpp:	Console()->Register("broadcast", "r", CFGFLAG_SERVER, ConBroadcast, this, "Broadcast message");
src/game/server/gamecontext.cpp:	Console()->Register("add_vote", "sr", CFGFLAG_SERVER, ConAddVote, this, "Add a voting option");
src/game/server/gamecontext.cpp:	Console()->Register("vote", "r", CFGFLAG_SERVER, ConVote, this, "Force a vote to yes/no");
src/engine/client/client.cpp:	m_pConsole->Register("rcon", "r", CFGFLAG_CLIENT, Con_Rcon, this, "Send specified command to rcon");
src/engine/client/client.cpp:	m_pConsole->Register("play", "r", CFGFLAG_CLIENT|CFGFLAG_STORE, Con_Play, this, "Play the file specified");
src/engine/server/server.cpp:	Console()->Register("ban", "s?ir", CFGFLAG_SERVER|CFGFLAG_STORE, ConBanExt, this, "Ban player with ip/client id for x minutes for any reason");
src/engine/server/server.cpp:	Console()->Register("kick", "i?r", CFGFLAG_SERVER, ConKick, this, "Kick player with specified id for any reason");
src/engine/shared/netban.cpp:	Console()->Register("ban", "s?ir", CFGFLAG_SERVER|CFGFLAG_MASTER|CFGFLAG_STORE, ConBan, this, "Ban ip for x minutes for any reason");
src/engine/shared/netban.cpp:	Console()->Register("ban_range", "ss?ir", CFGFLAG_SERVER|CFGFLAG_MASTER|CFGFLAG_STORE, ConBanRange, this, "Ban ip range for x minutes for any reason");
src/engine/shared/console.cpp:	Register("echo", "r", CFGFLAG_SERVER|CFGFLAG_CLIENT, Con_Echo, this, "Echo the text");
src/engine/shared/console.cpp:	Register("exec", "r", CFGFLAG_SERVER|CFGFLAG_CLIENT, Con_Exec, this, "Execute the specified file");
src/game/client/components/binds.cpp:	Console()->Register("bind", "sr", CFGFLAG_CLIENT, ConBind, this, "Bind key to execute the command");
src/game/client/components/chat.cpp:	Console()->Register("say", "r", CFGFLAG_CLIENT, ConSay, this, "Say in chat");
src/game/client/components/chat.cpp:	Console()->Register("say_team", "r", CFGFLAG_CLIENT, ConSayTeam, this, "Say in team chat");
src/game/client/components/chat.cpp:	Console()->Register("whisper", "ir", CFGFLAG_CLIENT, ConWhisper, this, "Whisper to a client in chat");
src/game/client/components/voting.cpp:	Console()->Register("vote", "r", CFGFLAG_CLIENT, ConVote, this, "Vote yes/no");
```

In all of these cases, the change either makes something more intuitive or doesn't have an impact at all.